api/src/accounts/AccountService.ts

```diff
@@ -44,4 +54,119 @@ export default class AccountService {
             );
         }
     }
+
+    public async getAccountByProduct(
+        org: Org,
+        product: AccountProduct,
+    ): Promise<DataManagerAccount | TagManagerAccount> {
+        if (product === AccountProduct.TAG_MANAGER)
+            return await this.repoFactory<TagManagerAccountRepo>(TagManagerAccount).getFromOrg(org);
+        if (product === AccountProduct.DATA_MANAGER) {
+            return await this.repoFactory<DataManagerAccountRepo>(DataManagerAccount).getFromOrg(
+                org,
+            );
+        }
+        throw new GenericError(
+            `Account for product ${product} not implemented.`,
+            LogPriority.ERROR,
+        );
+    }
+
+    public async unsubscribeAccount(
+        org: Org,
+        account: TagManagerAccount | DataManagerAccount | undefined,
```

Comment:
why are we testing for undefined?

Reply:
moving the check to this method will simplify the call:
`                return this.accountService.unsubscribeAccount(
                    org,
                    await this.accountService.getAccountByProduct(org, data.product),
                );`
                
if we want to check at the moment we call we need to use a support constant and throw before the call.
It just look cleaner this way IMO.

Reply:
Alternatively we need a _getAccountByProductThrows_

Reply:
Fixed